test/site/get-site-status-by-exporter-name.api-test.ts

```diff
@@ -0,0 +1,129 @@
+        .mockSuccessfulExpandCallWithExpandString(expand)
+        .mockSuccessfulFilterCallWithFilterString(filter)
+        .mockSuccessfulGraphGetCall(graphGetSiteStatusResponseDto);
+    },
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) =>
+      api.getWithoutAuth(`/api/v1/sites?exporterName=${siteStatusByExporterNameQueryDto.exporterName}`, incorrectAuth?.headerName, incorrectAuth?.headerValue),
+  });
+
+  withCommonGraphExceptionHandlingTests({
+    givenRequestWouldOtherwiseSucceed: () => {
+      mockGraphClientService
+        .mockSuccessfulGraphApiCallWithPath(path)
+        .mockSuccessfulExpandCallWithExpandString(expand)
+        .mockSuccessfulFilterCallWithFilterString(filter);
+    },
+    givenGraphServiceCallWillThrowError: (error: Error) => {
+      mockGraphClientService.mockUnsuccessfulGraphGetCall(error);
+    },
+    makeRequest: () => api.get(`/api/v1/sites?exporterName=${siteStatusByExporterNameQueryDto.exporterName}`),
+  });
+
+  const statusCodeTestInputs = [
+    {
+      siteStatus: 'Created',
+      expectedStatusCode: 200,
+    },
+    {
+      siteStatus: 'Provisioning',
+      expectedStatusCode: 202,
+    },
+    {
+      siteStatus: 'Failed',
+      expectedStatusCode: 424,
+    },
+  ];
+  it.each(statusCodeTestInputs)('returns $expectedStatusCode if graph replies with $siteStatus', async ({ siteStatus, expectedStatusCode }) => {
+    const {
+      siteStatusByExporterNameQueryDto: modifiedSiteStatusByExporterNameQueryDto,
+      siteStatusByExporterNameResponse: modifiedSiteStatusByExporterNameResponse,
+      graphServiceGetParams: { path: modifiedPath, expand: modifiedExpand, filter: modifiedFilter },
+      graphGetSiteStatusResponseDto: modifiedGraphGetSiteStatusResponseDto,
+    } = new getSiteStatusByExporterNameGenerator(valueGenerator).generate({ numberToGenerate: 1, status: siteStatus });
+
+    mockGraphClientService
+      .mockSuccessfulGraphApiCallWithPath(modifiedPath)
+      .mockSuccessfulExpandCallWithExpandString(modifiedExpand)
+      .mockSuccessfulFilterCallWithFilterString(modifiedFilter)
+      .mockSuccessfulGraphGetCall(modifiedGraphGetSiteStatusResponseDto);
+
+    const { status, body } = await api.get(`/api/v1/sites?exporterName=${modifiedSiteStatusByExporterNameQueryDto.exporterName}`);
+
+    expect(status).toBe(expectedStatusCode);
+    expect(body).toStrictEqual(modifiedSiteStatusByExporterNameResponse);
+  });
+
+  it('returns 404 with an empty object response if the site does not exist in sharepoint', async () => {
+    mockGraphClientService
+      .mockSuccessfulGraphApiCallWithPath(path)
+      .mockSuccessfulExpandCallWithExpandString(expand)
+      .mockSuccessfulFilterCallWithFilterString(filter)
+      .mockSuccessfulGraphGetCall({ value: [] });
+
+    const { status, body } = await api.get(`/api/v1/sites?exporterName=${siteStatusByExporterNameQueryDto.exporterName}`);
+
+    expect(status).toBe(404);
+    expect(body).toStrictEqual({});
```

Comment:
Why is the response body empty in this case? Do we not have a "Not found" / "Not Found" message for the user?